src/goto-symex/goto_symex.cpp

```diff
@@ -87,8 +96,294 @@ void goto_symext::symex_assign(statet &state, const code_assignt &code)
     if(state.source.pc->source_location.get_hide())
       assignment_type = symex_targett::assignment_typet::HIDDEN;
 
+    symex_assignt symex_assign{
+      state, assignment_type, ns, symex_config, target};
+
+    // Try to constant propagate potential side effects of the assignment, when
+    // simplification is turned on and there is one thread only. Constant
+    // propagation is only sound for sequential code as here we do not take into
+    // account potential writes from other threads when propagating the values.
+    if(symex_config.simplify_opt && state.threads.size() <= 1)
+    {
+      if(constant_propagate_assignment_with_side_effects(
+           state, symex_assign, lhs, rhs))
+        return;
+    }
+
     exprt::operandst lhs_if_then_else_conditions;
-    symex_assignt{state, assignment_type, ns, symex_config, target}.assign_rec(
+    symex_assign.assign_rec(
       lhs, expr_skeletont{}, rhs, lhs_if_then_else_conditions);
   }
 }
+
+/// Maps the given array expression containing constant characters to a string
+/// containing only alphanumeric characters
+///
+/// \param char_array: array_exprt containing characters represented by
+///   expressions of kind constant_exprt and type unsignedbv_typet or
+///   signedbv_typet
+/// \return a string containing only alphanumeric characters
+static std::string get_alnum_string(const array_exprt &char_array)
+{
+  const auto &ibv_type =
+    to_integer_bitvector_type(to_array_type(char_array.type()).subtype());
+
+  const std::size_t n_bits = ibv_type.get_width();
```

Comment:
:pick: `ibv_type` is used only once, it can be inlined here since the variable name doesn't give any information.

Reply:
Does provide information, and is more readable as is imo

Reply:
I wouldn't know what `ibv` meant if it wasn't followed by `to_integer_bitvector_type`, so at least to me it gives no information